queue/config.go

```diff
@@ -0,0 +1,38 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package queue
+
+import (
+	"fmt"
+	"time"
+
+	diskqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/diskqueue"
+	memqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/memqueue"
+)
+
+type Config struct {
+	Type         string             `config:"type"`
+	MemSettings  memqueue.Settings  `config:"memqueue"`
+	DiskSettings diskqueue.Settings `config:"diskqueue"`
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Type: "memory",
+		MemSettings: memqueue.Settings{
+			Events:         1024,
+			FlushMinEvents: 256,
+			FlushTimeout:   5 * time.Millisecond,
+		}, //memqueue should have a DefaultSettings()
+		DiskSettings: diskqueue.DefaultSettings(),
+	}
+}
+
+func (c *Config) Validate() error {
+	if c.Type != "memory" && c.Type != "disk" {
```

Comment:
Should we validate or warn when the settings don't match the type? like if MemSettings were provided when Type is "disk"?